peachjam/forms.py

```diff
@@ -188,6 +188,15 @@ def _save_m2m(self):
                 self.instance.file_as_pdf.delete()
                 self.instance.ensure_file_as_pdf()
 
+    def clean(self):
+        cleaned_data = super().clean()
+        if "file" in self.changed_data:
+            # clear all the fields that are derived from the file
+            for field in ["mimetype", "size", "file_as_pdf", "filename"]:
+                self.instance.__dict__[field] = None
```

Comment:
`setattr` is the preferred method: `setattr(self.instance, field, None)`

Reply:
that is exactly what `self.instance.field = None` does

Reply:
or just do it explicitly: `self.instance.mimetype = None` etc. :)

Reply:
thanks, updated